### ISSUE TITLE
How to explain Scikit-learn logistic regression binary classification probability output using TimeSHAP?

### ISSUE THREAD

**Issue by jagandecapri** [feedzai/timeshap#13]
Hi,

Thank you for the amazing work done in this repository. 🙂 

In TimeSHAP documentation, it is written that the output that needs to be fed to TimeSHAP is of the shape `(#sequences; 1)`. The PyTorch [example](https://github.com/feedzai/timeshap/blob/main/notebooks/AReM/AReM.ipynb) uses a `Linear` layer with output dimension 1 and a `Sigmoid` layer at the end to generate a single value that conforms to TimeSHAP requirement.

I have a binary classification model where there is some initial processing done to a 3-D NumPy input, `(#sequences; #sequence length; #features)`. In the last step, [Scikit-learn Logistic Regression](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html#sklearn.linear_model.LogisticRegression) is used to make the predictions. The predictions are made through the `predict_proba` function in the Logistic Regression class. It returns `(#sequences, 2)` output where the probabilities of both the classes are returned as an array of two values where each row sums to a probability of 1. For instance, for sequence length of 3, the probabilities could be as following,

```
[
[0.3, 0.7],
[0.6, 0.4],
[0.2, 0.8],
]
```

How can the Scikit-learn Logistic Regression's probabilities be used for an explanation using TimeSHAP?

Thank you.

**Comment by JoaoPBSousa** [comment]
Hi @jagandecapri,

From what you described, I think the best option is to wrap the Scikit-learn Logistic Regression model with a Wrapper. We briefly covered them in our README [here](https://github.com/feedzai/timeshap#model-wrappers).
This wrapper will function as the mediator between the model and TimeSHAP, and will allow you to take the output of the model with any arbitrary shape and select the specific probability you want to return to TimeSHAP.

In the example you describe, imagine you want to explain your model when predicting the score of class 1 for the last element of the sequence.  The wrapper would take the model output you described, and just return `0.8` to TimeSHAP.

In case you implement this wrapper and want to share it, we can make it available on the repo under the wrappers [directory](https://github.com/feedzai/timeshap/tree/main/src/timeshap/wrappers).

Let me know if you have any further questions.

**Comment by jagandecapri** [comment]
Hi @JoaoPBSousa, noted. Thank you.
